cmd/soroban-rpc/internal/xdr2json/conversion.go

```diff
@@ -0,0 +1,166 @@
+//nolint:lll
+package xdr2json
+
+/*
+// See preflight.go for add'l explanations:
+
+#include <stdlib.h>
+#include "../../lib/xdrjson.h"
+
+#cgo windows,amd64 LDFLAGS: -L${SRCDIR}/../../../../target/x86_64-pc-windows-gnu/release-with-panic-unwind/ -lpreflight -lntdll -static -lws2_32 -lbcrypt -luserenv
+#cgo darwin,amd64  LDFLAGS: -L${SRCDIR}/../../../../target/x86_64-apple-darwin/release-with-panic-unwind/ -lpreflight -ldl -lm
+#cgo darwin,arm64  LDFLAGS: -L${SRCDIR}/../../../../target/aarch64-apple-darwin/release-with-panic-unwind/ -lpreflight -ldl -lm
+#cgo linux,amd64   LDFLAGS: -L${SRCDIR}/../../../../target/x86_64-unknown-linux-gnu/release-with-panic-unwind/ -lpreflight -ldl -lm
+#cgo linux,arm64   LDFLAGS: -L${SRCDIR}/../../../../target/aarch64-unknown-linux-gnu/release-with-panic-unwind/ -lpreflight -ldl -lm
+*/
+import "C"
+
+import (
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"unsafe"
+
+	"github.com/pkg/errors"
+
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+)
+
+const (
+	FormatBase64 = "base64"
+	FormatJSON   = "json"
+)
+
+var errInvalidFormat = fmt.Errorf(
+	"expected %s for optional 'xdrFormat'",
+	strings.Join([]string{FormatBase64, FormatJSON}, ", "))
+
+// Convert takes an XDR object (`xdr`) and its serialized bytes (`field`)
+// and returns the JSON-formatted serialization of that object.
+//
+// The `xdr` object does not need to actually be initialized/valid:
+// we only use it to determine the name of the structure. We could just
+// accept a string, but that would make mistakes likelier than passing the
+// structure itself (by reference).
+func Convert(xdr interface{}, field []byte) (map[string]interface{}, error) {
+	xdrTypeName := reflect.TypeOf(xdr).Name()
+	goStr := convertStr(xdrTypeName, field)
+	return jsonify(goStr)
+}
+
+// ConvertAny takes a valid XDR object (`xdr`) and returns a
+// JSON-formatted serialization of that object.
+//
+// Unlike `Convert`, the value here needs to be valid and
+// serializable.
+func ConvertAny(xdr interface{}) (map[string]interface{}, error) {
```

Comment:
`ConvertAny` is ambiguous, it kinda overlaps semantics of `Convert` also, for consideration, how about rename:
`Convert` -> `ConvertBinary`
`ConvertAny` -> `ConvertInstance`

Reply:
Nice, yeah! Didn't give the naming too much thought so thanks for the "fresh eyes" perspective :+1: